src/java/org/apache/cassandra/service/accord/AccordKeyspace.java

```diff
@@ -0,0 +1,843 @@
+
+        static final Columns statics = Columns.from(Lists.newArrayList(max_timestamp, last_executed_timestamp, last_executed_micros, last_write_timestamp, blind_witnessed));
+        static final Columns regulars = Columns.from(Lists.newArrayList(data));
+        private static final RegularAndStaticColumns all = new RegularAndStaticColumns(statics, regulars);
+        private static final RegularAndStaticColumns justStatic = new RegularAndStaticColumns(statics, Columns.NONE);
+        private static final RegularAndStaticColumns justRegular = new RegularAndStaticColumns(Columns.NONE, regulars);
+
+        static boolean hasStaticChanges(AccordCommandsForKey commandsForKey)
+        {
+            return commandsForKey.maxTimestamp.hasModifications()
+                   || commandsForKey.lastExecutedTimestamp.hasModifications()
+                   || commandsForKey.lastExecutedMicros.hasModifications()
+                   || commandsForKey.blindWitnessed.hasModifications();
+        }
+
+        static RegularAndStaticColumns columnsFor(AccordCommandsForKey commandsForKey)
+        {
+            boolean hasStaticChanges = hasStaticChanges(commandsForKey);
+            boolean hasRegularChanges = commandsForKey.uncommitted.map.hasAdditions()
+                                        || commandsForKey.committedById.map.hasAdditions()
+                                        || commandsForKey.committedByExecuteAt.map.hasAdditions();
+
+            if (hasStaticChanges && hasRegularChanges)
+                return all;
+            else if (hasStaticChanges)
+                return justStatic;
+            else if (hasRegularChanges)
+                return justRegular;
+            else
+                throw new IllegalArgumentException();
+        }
+    }
+
+    private static TableMetadata parse(String name, String description, String cql)
+    {
+        return CreateTableStatement.parse(format(cql, name), ACCORD_KEYSPACE_NAME)
+                                   .id(TableId.forSystemTable(ACCORD_KEYSPACE_NAME, name))
+                                   .comment(description)
+                                   .gcGraceSeconds((int) TimeUnit.DAYS.toSeconds(90))
+                                   .build();
+    }
+
+    public static KeyspaceMetadata metadata()
+    {
+        return KeyspaceMetadata.create(ACCORD_KEYSPACE_NAME, KeyspaceParams.local(), tables(), Views.none(), Types.none(), Functions.none());
+    }
+
+    private static Tables tables()
+    {
+        return Tables.of(Commands, CommandsForKey);
+    }
+
+    private static <T> ByteBuffer serialize(T obj, IVersionedSerializer<T> serializer, int version) throws IOException
+    {
+        int size = (int) serializer.serializedSize(obj, version);
+        try (DataOutputBuffer out = new DataOutputBuffer(size))
+        {
+            serializer.serialize(obj, out, version);
+            assert size == out.buffer().limit();
```

Comment:
`out.buffer(false)`, we duplicate by default and we don't need to allocate here.

Also, should save the result as we need to get access here and line 285

Reply:
also useful error message plz!